### PR TITLE
feat: use dyndata endpoint for drops

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Related: https://github.com/kodadot/private-workers/pull/188
- [x] Requires deployment https://github.com/kodadot/private-workers/pull/188

> [!CAUTION]
> ‼️ Please test on ahk collection only!

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### CODE REVIEW

**Comment by preschian** [comment]
hi, can we merge this one first, before this PR: https://github.com/kodadot/nft-gallery/pull/10670?

some changes probably conflict, such as `useUpdateMetadata`

**Comment by preschian** [comment]
Fyi: there is a case when sn probably returns not in ordered number when there is a mixing while minting simultaneously between the canary and production environment. But no worries, the quick fix is just to update metadata as usual. and once the pr is deployed to production, it should be fine

**Comment by hassnian** [comment]
canary doesn't have this 

CleanShot 2024-07-30 at 14 30 23

**Review by hassnian — APPROVED** [review]

File: components/collection/drop/modal/shared/SuccessfulDrop.vue [hunk 1]
```diff
@@ -41,19 +42,34 @@ const mintedNft = computed<MintedNFT | undefined>(
...
     name: item.name,
     image: item.image,
     collection: item.collection.id,
     collectionName: item.collection.name,
     mimeType: item.mimeType,
-  })),
-)
+    metadata: item.metadata,
+  })
+}
```
[the remark this answers is not in the document]
Comment (preschian) [reply components/collection/drop/modal/shared/SuccessfulDrop.vue]: this one also works. as long as `items = ref()` instead of `items = computed()`

**Comment by preschian** [comment]
Hi, lmk what you think. This is an attempt to make mint drops on the assethub faster. ref: https://github.com/kodadot/private-workers/tree/main/services/dyndata#faster-serial-number

**Comment by vikiival** [comment]
cc @hassnian, @Jarsen136

**Review by Jarsen136 — APPROVED** [review]
wfm

File: composables/drop/useGenerativeDropMint.ts [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
+import { type DoResult, updateMetadata } from '@/services/fxart'
 import { useDrop } from '@/components/drops/useDrops'
 import unlockableCollectionById from '@/queries/subsquid/general/unlockableCollectionById.graphql'
 import { FALLBACK_DROP_COLLECTION_MAX } from '@/utils/drop'
-import type {
-  MassMintNFT,
-} from '@/composables/drop/massmint/useDropMassMint'
-import useDropMassMint from '@/composables/drop/massmint/useDropMassMint'
 import useDropMassMintListing from '@/composables/drop/massmint/useDropMassMintListing'
+import type { MintedNFT } from '~/components/collection/drop/types'
```
Comment (hassnian) [inline composables/drop/useGenerativeDropMint.ts:7]:
```suggestion
import type { MintedNFT } from '@/components/collection/drop/types'
```

**Review by hassnian — APPROVED** [review]

**Pull request merged**